File: frontend/src/services/lecturer-dashboard.service.ts
import http from '@/lib/http'
import { type ApiResponse } from '@/schemas/base.schema'
import type { LecturerDashboardResponse } from '@/schemas/lecturer-dashboard.schema'

export const lecturerDashboardService = {
  getLecturerDashboard: async (params: { fromDate: string; toDate: string }) => {
    const response = await http.get<ApiResponse<LecturerDashboardResponse>>('/lecturer/dashboard', {
      params
    })
    return response.data.data
  }
}

File: backend/src/main/java/iuh/labbooking/service/dashboard/LecturerDashboardService.java
package iuh.labbooking.service.dashboard;

import iuh.labbooking.dto.response.dashboard.lecturer.LecturerDashboardResponse;
import java.time.LocalDate;

public interface LecturerDashboardService {
    LecturerDashboardResponse getLecturerDashboard(LocalDate fromDate, LocalDate toDate);
}
